src/ring_theory/adjoin_root.lean

```diff
@@ -102,13 +102,25 @@ lemma algebra_map_eq' [comm_semiring S] [algebra S R] :
 
 variables {S}
 
+lemma finite_type : algebra.finite_type R (adjoin_root f) :=
+(algebra.finite_type.polynomial R).of_surjective _ (ideal.quotient.mkₐ_surjective R _)
```

Comment:
This lemma could also be removed (and the same is true for the lemma `finite_type R R[X]`) because it follows from the `finite_presentation` version below using [algebra.finite_type.of_finite_presentation](https://leanprover-community.github.io/mathlib_docs/ring_theory/finiteness.html#algebra.finite_type.of_finite_presentation). I would remove them if people think that's better, or I could change the proofs to use `algebra.finite_type.of_finite_presentation` (but the `finite_type R R[X]` result will then need to be moved down).